photutils/psf/models.py

```diff
@@ -4,17 +4,428 @@
 from ..extern.nddata_compat import extract_array
 
+__all__ = ['FittableImageModel2D', 'NonNormalizable',
+           'IntegratedGaussianPRF', 'PRFAdapter',
+           'prepare_psf_model', 'get_grouped_psf_model']
 
-__all__ = ['IntegratedGaussianPRF', 'PRFAdapter', 'prepare_psf_model',
-           'get_grouped_psf_model']
+
+class NonNormalizable(Warning):
+    """
+    Used to undicate that a :py:class:`FittableImageModel2D` model is
+    non-normalizable.
+
+    """
+    pass
+
+
+class FittableImageModel2D(Fittable2DModel):
+    """
+    A fittable 2D model of an image allowing for image intensity scaling
+    and image translations.
+
+    This class takes 2D image data and computes the
+    values of the model at arbitrary locations (including at intra-pixel,
+    fractional positions) within this image using spline interpolation
+    provided by :py:class:`~scipy.interpolate.RectBivariateSpline`.
+
+    The fittable model provided by this class has three model parameters:
+    an image intensity scaling factor (`flux`) which is applied to
+    (normalized) image, and two positional parameters (`x_0` and `y_0`)
+    indicating the location of a feature in the coordinate grid on which
+    the model is to be evaluated.
+
+    If this class is initialized with `flux` (intensity scaling factor)
+    set to `None`, then `flux` is be estimated as ``|sum(data)|``.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Array containing 2D image.
+
+    origin : tuple, None, optional
+        A reference point in the input image ``data`` array. When origin is
+        `None`, origin will be set at the middle of the image array.
+
+        If `origin` represents the location of a feature (e.g., the position
+        of an intensity peak) in the input ``data``, then model parameters
+        `x_0` and `y_0` show the location of this peak in an another target
+        image to which this model was fitted. Fundamentally, it is the
+        coordinate in the model's image data that should map to
+        coordinate (`x_0`, `y_0`) of the output coordinate system on which the
+        model is evaluated.
+
+        Alternatively, when `origin` is set to ``(0,0)``, then model parameters
+        `x_0` and `y_0` are shifts by which model's image should be translated
+        in order to match a target image.
+
+    normalize : bool, optional
+        Indicates whether or not the model should be build on normalized
+        input image data. If true, then the normalization constant (*N*) is
+        computed so that
+
+        .. math::
+            N \cdot C \cdot |\Sigma_{i,j}D_{i,j}| = 1,
+
+        where *N* is the normalization constant, *C* is correction factor
+        given by the parameter ``correction_factor``, and :math:`D_{i,j}` are
+        the elements of the input image ``data`` array.
+
+    correction_factor : float, optional
+        A strictly positive number that represents correction that needs to
```

Comment:
This sounds in line with your expectation: _"I thought this was the correction factor in the sense that if it's bigger, the output value (at fixed flux) would be bigger."_

Finally, I do not want to have a very specific naming is not too specific, especially because it has nothing to do with the ``flux`` (as model's parameter) but rather it is a correction factor that takes into account the fact that **actual** normalization of model's data may be off compared to **"true"** normalization due to the fact that some data (="flux" - do not confuse with the ``flux`` model parameter) have not been included into the normalization because they were outside the image and thus, have not contributed to the computation of the normalization factor.

Last paragraph may be not very clear so I'll try to provide a new angle: Correction factor corrects the total flux of model's data (by which data is normalized) and not model's ``flux`` parameter though the correction of the model will result in corrections to fitted parameters.

``correction_factor`` is generic enough to be interpreted however one wishes to interpret it (as long as it is documented what it is).